ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -134,32 +136,45 @@ def _router_bgp_vrfs(self) -> dict | None:
                     raise AristaAvdMissingVariableError(f"'vrf_id' or 'vrf_vni' for VRF '{vrf_name}")
 
                 leaf_overlay_rt = f"{self._rt_admin_subfield or bgp_vrf_id}:{bgp_vrf_id}"
-                route_targets = {
-                    "import": {af: [leaf_overlay_rt] for af in vrf_address_families},
-                    "export": {af: [leaf_overlay_rt] for af in vrf_address_families},
-                }
+
+                route_targets = {"import": [], "export": []}
+
+                for af in vrf_address_families:
+                    route_target = {"address_family": af, "route_targets": [leaf_overlay_rt]}
+                    for key, value in route_targets.items():
```

Comment:
IMO the for loop is not making it easier to read. I suggest two checks for import and export.